src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs

```diff
@@ -89,6 +89,39 @@ public async Task<Response<ConfigurationSetting>> AddAsync(ConfigurationSetting
             }
         }
 
+        public async Task<Response<ConfigurationSetting>> AddAsync(string key, string value, CancellationToken cancellation = default)
+        {
+            if (string.IsNullOrEmpty(key)) throw new ArgumentNullException($"{nameof(key)}");
+            if (string.IsNullOrEmpty(value)) throw new ArgumentNullException($"{nameof(value)}");
+
+            Uri uri = BuildUriForKvRoute(key);
+
+            using (HttpMessage message = _pipeline.CreateMessage(cancellation))
+            {
+                ReadOnlyMemory<byte> content = Serialize(value);
+                message.SetRequestLine(HttpVerb.Put, uri);
+
+                message.AddHeader("Host", uri.Host);
+                message.AddHeader(IfNoneMatch, "*");
+                message.AddHeader(MediaTypeKeyValueApplicationHeader);
+                message.AddHeader(HttpHeader.Common.JsonContentType);
+                message.AddHeader(HttpHeader.Common.CreateContentLength(content.Length));
+                AddClientRequestID(message);
+                AddAuthenticationHeaders(message, uri, HttpVerb.Put, content, _secret, _credential);
+
+                message.SetContent(HttpMessageContent.Create(content));
+
+                await _pipeline.SendMessageAsync(message).ConfigureAwait(false);
+
+                var response = message.Response;
+                if (response.Status == 200)
+                {
+                    return await CreateResponse(response, cancellation);
+                }
+                else throw new RequestFailedException(response);
```

Comment:
Do you want to use a separate exception type and/or message for 412/precondition failed? (e.g. "The resource already exists :(")